src/main/webapp/js/chart-loader.js

```diff
@@ -1,26 +1,48 @@
  google.charts.load('current', {packages: ['corechart']});
  google.charts.setOnLoadCallback(drawChart);
- function drawChart(){
-
-    var book_data = new google.visualization.DataTable();
-    //define columns for the DataTable instance
-    book_data.addColumn('string', 'Book Title');
-    book_data.addColumn('number', 'Votes');
-
-    //add data to book_data
-    book_data.addRows([
-    ["The Best We Could Do", 6],
-    ["Sing, Unburied, Sing", 10],
-    ["The Book of Unknown Americans", 7],
-    ["The 57 Bus", 4],
-    ["The Handmaid's Tale", 8]
-    ]);
-
-    var chart = new google.visualization.BarChart(document.getElementById('book_chart'));
+
+ // draw the chart from dataTable
+ function drawChart(dataTable){
+
+    var chart = new google.visualization.BarChart(document.getElementById('chart'));
     var chart_options = {
     width: 800,
     height: 400
     };
-    chart.draw(book_data, chart_options);
 
-  }
\ No newline at end of file
+    chart.draw(dataTable, chart_options);
+
+  }
+
+ // fetches data to be used in the chart.
+ function fetchMessageData() {
+          fetch("/messagechart")
+               .then((response) => {
+                  return response.json();
+               })
+               .then((msgJson) => {
+                 var msgData = new google.visualization.DataTable();
+                 //define columns for the DataTable instance
+                 msgData.addColumn('date', 'Date');
+                 msgData.addColumn('number', 'Message Count');
+
+
+                 for (i = 0; i < msgJson.length; i++) {
+                     msgRow = [];
+                     var timestampAsDate = new Date (msgJson[i].timestamp);
+                     var totalMessages = i + 1;
+
+                     msgRow.push(timestampAsDate);
+                     msgRow.push(totalMessages);
+
+                     //console.log(msgRow);
+                     msgData.addRow(msgRow);
+
+                 }
+                 //console.log(msgData);
+                 drawChart(msgData);
```

Comment:
In the project chat, we're discussing `drawChart` appearing twice. Please follow the discussion and try to figure out if this is supposed to occur - and if not, replace `drawChart` with what it's supposed to be instead.